spec/models/user_spec.rb

```diff
@@ -137,4 +137,115 @@
       expect(user.list_groups.include?(groups.first.name)).to be_truthy
     end
   end
+
+  context "Class Methods" do
+    context "#search_all" do
+      it "returns users by partial search terms" do
+        # cohort_1 = create(:cohort, name: "1608")
+        # cohort_2 = create(:cohort, name: "1610")
+        # cohort_3 = create(:cohort, name: "1701")
+
+        role_1 = create(:role, name: "mentor")
+        role_2 = create(:role, name: "enrolled")
+        role_3 = create(:role, name: "exited")
+
+        users = create_list(:user, 3)
+        # users.first.cohort = cohort_1
+        # users.second.cohort = cohort_2
+        # users.third.cohort = cohort_3
+
+        users.first.roles << role_1
+        users.second.roles << role_2
+        users.third.roles << role_3
+
+        group_1 = create(:group, users: [users.first], name: "grouping")
+        group_2 = create(:group, users: [users.second], name: "groupers")
+        group_2 = create(:group, users: [users.third], name: "having")
+
+
+        # search_return = User.search_all('16')
+        #
+        # expect(search_return).to include(users.first)
+        # expect(search_return).to include(users.second)
+
+        search_return = User.search_all("ed")
+
+        expect(search_return).to include(users.second)
+        expect(search_return).to include(users.third)
+
+        search_return = User.search_all("ing")
+
+        expect(search_return).to include(users.first)
+        expect(search_return).to include(users.third)
+      end
+    end
+    #
+    # context "#search_cohorts" do
+    #   it "returns users by partial cohort search terms" do
```

Comment:
Do we care about partial name search?

Reply:
Hmm, I suppose that is good to ask. We definitely want partial name search for users, groups, etc. I'm not as adamant about partial name search for cohorts. If you typed in a partial name of a cohort say `16` you'd get everyone enrolled that year. I could see it being helpful if you wanted to get all front and back students for a cohort. So if you typed 1801 you would get all students for both front and back which would be helpful.

Reply:
:+1: should be too hard to add another method similar to `find_by_name` that uses a regexp instead of `==`. Add a trello card?